lib/books.ts

```diff
@@ -0,0 +1,111 @@
+import { create } from "zustand";
+import { immer } from "zustand/middleware/immer";
+
+
+// NOTE: VV = versions & variants
+// sources affect visibility of versions, versions affect visibility of variants
+
+interface BookState {
+  enabledBooks: any;
+  toggleEnabledBook: (index: number) => void;
+  selectedDocumentVVs: any; 
+  setDocumentVV: (docName:string, type:string, vv:string) => void;
+  currentDocument: string;
+  setCurrentDocument: (docName:string) => void;
+}
+
+export const useBookStore = create<BookState, [["zustand/immer", never]]>(
+  immer((set,get) => {
+
+    const initialBooks = [
+      ["DATP","Dark Alleys & Twisted Paths",true],
+      ["DPAS","Dark Pacts & Ancient Secrets",true],
+    ];
+
+    const localBooks = typeof localStorage !== 'undefined' ? JSON.parse(localStorage.getItem("books")) : null;
+    
+    const joinedBooks = localBooks ? 
+      initialBooks.map( (book) => 
+        [ book[0], book[1], 
+          localBooks.find((elem) => elem[0]==book[0]) ?.at(2) ?? book[2]]
+      ) : initialBooks;
+
+    // TODO: setting this properly, lol
+    setTimeout( () => (set(state => { state.enabledBooks = joinedBooks })), 1000 )
```

Comment:
This is the makeshift hack to ignore hydration errors we discussed @sean-clayton .
So far I have been unable to make hydration work, this definitely needs to be solved one way or another before merging.